### ISSUE TITLE
Visual script cannot create a variable of type packed scene.

### ISSUE THREAD

**Issue by fire** [godotengine/godot#15244]
**Godot version:**
master

**OS/device including version:**
windows 10

**Issue description:**
Visual script cannot create a variable of type packed scene. So I'm unable to use the scene in a variable. I should be able to use any type as a variable.

**Steps to reproduce:**

![godot windows opt tools 64_2018-01-01_07-41-09](https://user-images.githubusercontent.com/32321/34468875-28b75fb2-eec7-11e7-9eeb-a0af2390851c.png)

- [ ] I searched the existing [GitHub issues](https://github.com/godotengine/godot/issues?utf8=%E2%9C%93&q=is%3Aissue+) for potential duplicates.

**Comment by bojidar-bg** [comment]
Does it work if you select type = Object?

**Comment by fire** [comment]
I'm unable to select the scene from the bar in the scene export variables section. It is unselectable.

**Comment by fire** [comment]
It's greyed out.

**Comment by fire** [comment]
The proper type should be resource. I don't see any support for it.

**Comment by brainsick** [comment]
fire, would you happen to have a sample project available?  I'd love to see some Visual Script in context, kick the tires some.

I don't understand how you want to use the PackedScene.  At design time, I can just drag assets from the filesystem dock into the visual scripting grid.  These are inserted as Preload resources.  It seems that the default Variant variable can also hold resource; there's a ResourceType hint?

**Comment by fire** [comment]
This is the gdscript code:

```
extends WorldEnvironment

export (NodePath) var settings_vr
export (NodePath) var right_hand
export (NodePath) var desktop_settings_gui
export (NodePath) var root
export (PackedScene) var world

func _ready():	
	get_node(root).add_child(world.instance())

func start_vr():
	var interface = ARVRServer.find_interface("OpenVR")
	if !interface or !interface.initialize():
		return

	get_viewport().arvr = true
	get_viewport().hdr = false

	get_node(right_hand).connect("button_pressed", get_node(settings_vr), "_on_Right_Hand_button_pressed")

func _on_ConfirmButton_pressed():
	get_node(desktop_settings_gui).visible = false
	start_vr()
```

Notice `export (PackedScene) var world`.

Trying to convert this.

**Comment by fire** [comment]
Here's an example that also fails:

```
export (NativeScript) var openvr_render_model_native

func _ready():
	# instance our render model object
	ovr_render_model = openvr_render_model_native.new()
```

**Comment by fire** [comment]
<ibrahn> yeah, beyond that, I'm not sure what a good fix looks like. Just hacked ResourceType along to 17 and it does what you'd hope, but just flinging an updated list of strings into that box doesn't seem super maintainable

ibrahn discovered why.

![enum definition](https://i.imgur.com/jnNIwpA.png)

![editor definition](https://i.imgur.com/X9seCKf.png)

The enum and the enum list is broken.

**Comment by akien-mga** [comment]
> just flinging an updated list of strings into that box doesn't seem super maintainable

It's true, but I can't see of a simple way to do this automatically. I think the best solution for now is to update the list manually to match the enum, and add comments just before that list and the enum that they need to be kept in sync.

**Comment by fire** [comment]
```
p_list->push_back(PropertyInfo(Variant::INT, "hint", PROPERTY_HINT_ENUM, "None,Range,ExpRange,Enum,ExpEasing,Length,SpriteFrame,KeyAccel,Flags,Layers2dRender,Layers2dPhysics,Layer3dRender,Layer3dPhysics,File,Dir,GlobalFile,GlobalDir,ResourceType,MultilineText,ColorNoAlpha,ImageCompressLossy,ImageCompressLossLess,ObjectId,String"));
```

This is what I have. What areLayer3dRender etc. and ObjectId / String? There's also:

```
	PROPERTY_HINT_NODE_PATH_TO_EDITED_NODE, ///< so something else can provide this (used in scripts)
	PROPERTY_HINT_METHOD_OF_VARIANT_TYPE, ///< a method of a type
	PROPERTY_HINT_METHOD_OF_BASE_TYPE, ///< a method of a base type
	PROPERTY_HINT_METHOD_OF_INSTANCE, ///< a method of an instance
	PROPERTY_HINT_METHOD_OF_SCRIPT, ///< a method of a script & base
	PROPERTY_HINT_PROPERTY_OF_VARIANT_TYPE, ///< a property of a type
	PROPERTY_HINT_PROPERTY_OF_BASE_TYPE, ///< a property of a base type
	PROPERTY_HINT_PROPERTY_OF_INSTANCE, ///< a property of an instance
	PROPERTY_HINT_PROPERTY_OF_SCRIPT, ///< a property of a script & base
	PROPERTY_HINT_OBJECT_TOO_BIG, ///< object is too big to send
	PROPERTY_HINT_MAX,
```

**Comment by akien-mga** [comment]
Fixed by ef1c26a.
